src/common/vector_operations/vector_cast.cpp

```diff
@@ -136,6 +136,57 @@ static void string_cast_switch(Vector &source, Vector &result, SQLType source_ty
 	}
 }
 
+static void string_strict_cast_switch(Vector &source, Vector &result, SQLType source_type, SQLType target_type,
```

Comment:
Rather than duplicate the `string_cast_switch` here, I would prefer if you would pass the duckdb::StrictCast as a template parameter. i.e. you could make the string_cast_switch function as follows:

```cpp
template<class OP>
static void string_cast_switch...

...
UnaryExecutor::Execute<string_t, bool, OP, true>
...
```

And use it like this


```
string_cast_switch<duckdb::StrictCast>
// or
string_cast_switch<duckdb::Cast>
```

That way you can also quite easily merge the strict cast and the normal cast. You can have a wrapper in VectorOperations::Cast that looks like this:

```cpp
if (strict) {
    cast_switch<duckdb::StrictCast>(...)
} else {
    cast_switch<duckdb::Cast>(...)
}
```

Where the cast_switch has the current set of switches that are now present in VectorOperations::Cast, i.e.
```cpp
switch (source_type.id) {
case SQLTypeId::BOOLEAN:
	assert(source.type == TypeId::BOOL);
	numeric_cast_switch<bool, OP>(source, result, source_type, target_type, count);
	break;
...
```

Reply:
Changed that - this is indeed way prettier. Thanks!